### ISSUE TITLE
Style/RedundantLineContinuation: false positive

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#13337]
```
$ cat rubocop/contline.rb
# frozen_string_literal: true

def default_handler
  if (ctrl = CATEGORY_TO_CONTROLLER[arg[:category]]) \
     && (v = to_i(value))
    @res[ctrl] << v
  end
end
```
```
$ bundle exec rubocop -A rubocop/contline.rb
Inspecting 1 file
F

Offenses:

rubocop/contline.rb:4:54: C: [Corrected] Style/RedundantLineContinuation: Redundant line continuation.
  if (ctrl = CATEGORY_TO_CONTROLLER[arg[:category]]) \ ...
                                                     ^
rubocop/contline.rb:5:6: F: Lint/Syntax: unexpected token tAMPER
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
     && (v = to_i(value))
     ^

1 file inspected, 2 offenses detected, 1 offense corrected
```
```
$ cat rubocop/contline.rb
# frozen_string_literal: true

def default_handler
  if (ctrl = CATEGORY_TO_CONTROLLER[arg[:category]]) 
     && (v = to_i(value))
    @res[ctrl] << v
  end
end
```

--------

## Expected behavior

The backslash is needed, don't warn, don't remove it.

## Actual behavior

It is incorrectly considered superfluous and removed.  The updated code is wrong.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ bundle exec rubocop -V
1.67.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, analyzing as Ruby 2.7, running on ruby 3.3.5) [arm64-darwin23]
  - rubocop-performance 1.22.1
```
